packages/react-meteor-data/suspense/useFind.ts

```diff
@@ -0,0 +1,126 @@
+import { Meteor } from 'meteor/meteor'
+import { Mongo } from 'meteor/mongo'
+import { useEffect } from 'react'
+import isEqual from 'lodash/isEqual';
+import clone from 'lodash/clone'
+
+
+export const selectorsCache = new Map<Mongo.Collection<unknown>, SelectorEntry[]>();
+type SelectorEntry = {
+  findArgs: Parameters<Mongo.Collection<unknown>['find']>;
+  promise: Promise<unknown>;
+  result?: unknown;
+  error?: unknown;
+  counter: number;
+
+};
+export const removeFromArray =
+  <T>(list: T[], obj: T): void => {
+    if (obj) {
+      const index = list.indexOf(obj);
+      if (index !== -1) {
+        list.splice(index, 1);
+      }
+    }
+  }
+const useFindSuspense = <T = any>(
+  collection: Mongo.Collection<T>,
+  findArgs: Parameters<Mongo.Collection<T>['find']> | null,
+) => {
+  useEffect(() => {
+    const cachedSelectors = selectorsCache.get(collection) ?? [];
+    const _args = clone(findArgs)
+    const selector = cachedSelectors.find(x => isEqual(x.findArgs, _args));
+
+    if (selector) ++selector.counter;
+
+    return () => {
+      // defer
+      setTimeout(() => {
+        const cachedSelectors = selectorsCache.get(collection) ?? [];
+        const selector = cachedSelectors.find(x => isEqual(x.findArgs, _args));
+        if (selector && --selector.counter === 0) {
+          removeFromArray(cachedSelectors, selector);
+          selectorsCache.set(collection, cachedSelectors);
+        }
+      }, 0)
+    };
+  }, [findArgs, collection]);
+
+  useEffect(() => {
+    return () => {
+      // defer
+      setTimeout(() => {
+        const cachedSelectors = selectorsCache.get(collection) ?? [];
+        if (
+          findArgs === null && cachedSelectors.length > 0
+        ) {
+          for (const selector of cachedSelectors) {
+            removeFromArray(cachedSelectors, selector);
+          }
+          selectorsCache.set(collection, cachedSelectors);
+        }
+      }, 0)
+    }
+  }, [findArgs])
```

Comment:
It removes all of the selectors whenever one of them becomes `null` - that's incorrect. For example, if I have two components and only one of them becomes `null`, the other shouldn't be affected.

Reply:
Yeah, I could only think of using a key to solve this, but we would need three parameters...
As we talked about, cursors are not serializable.
I tried having a id being created every call of useFind but that was not helpful at all. Will try to push a useFind with this key ideia

Reply:
What is the cache trying to do? The collection system behind Meteor.find already has caching built in.

Reply:
Let's continue this discussion in https://github.com/meteor/react-packages/pull/387#issuecomment-1470815298.